src/core/ext/xds/xds_server_config_fetcher.cc

```diff
@@ -1092,19 +1092,19 @@ ServerConfigSelector::CallConfig XdsServerConfigFetcher::ListenerWatcher::
     FilterChainMatchManager::XdsServerConfigSelector::GetCallConfig(
         grpc_metadata_batch* metadata) {
   CallConfig call_config;
-  if (metadata->legacy_index()->named.path == nullptr) {
+  if (metadata->get_pointer(PathMetadata()) == nullptr) {
     call_config.error = GRPC_ERROR_CREATE_FROM_STATIC_STRING("No path found");
     return call_config;
   }
-  absl::string_view path = StringViewFromSlice(
-      GRPC_MDVALUE(metadata->legacy_index()->named.path->md));
-  if (metadata->legacy_index()->named.authority == nullptr) {
+  absl::string_view path =
+      metadata->get_pointer(PathMetadata())->as_string_view();
```

Comment:
Minimal to none... get_pointer boils down to:
`if ((present_fields_ & BIT_MASK_FOR_PATH) != 0) return &this->path_` (similarly for non-path things)

So there's no overhead if the compiler can prove present_fields_ is unchanged, and a test-and-branch otherwise.